### ISSUE TITLE
feat: Allow ApisixRoute's secretRef to provision nested configuration

### ISSUE THREAD

**Issue by lpiob** [apache/apisix-ingress-controller#1722]
### Description

Some plugins, like for example `aws-lambda` require to enter credentials in the ApisixRoute object:

```
      plugins:
        - name: aws-lambda
          config:
            authorization:
              iam:
                accesskey: "xxx"
                secretkey: "xxx"
```

We can fetch plugin configuration from secrets (proposed in #1408 and implemented in #1486) but current implementation does not support nested levels. 

I can use secretRef to set `config.accesskey` but not `config.authorization.iam.accesskey`.

Also, the secretRef in ApisixRoute behaves differently than secretRef in ApisixConsumer, where the secret [can be inserted at any point](https://apisix.apache.org/docs/ingress-controller/next/tutorials/enable-authentication-and-restriction/#key-auth-reference-secret-object).

I'd like to be able to set the nested plugin configuration or to use the secretRef in the same way I can do in ApisixConsumer.

**Comment by tao12345666333** [comment]
Good point.  Let me check it.

**Comment by ecoupal-believe** [comment]
Any news on that one ?

**Comment by ErikThorsell** [comment]
Would just like to add that this is relevant for the openid-connect plugin too.

```yaml
apiVersion: apisix.apache.org/v2
kind: ApisixPluginConfig
metadata:
  name: apisix-plugin-openid-authz
spec:
  plugins:
  - name: openid-connect
    enable: true
    config:
      client_id: "client"
      client_secret: <this-one-gets-subbed-just-fine>
      discovery: "http://keycloak:8080/auth/realms/mydomain/.well-known/openid-configuration"
      bearer_only: false
      access_token_in_authorization_header: true
      realm: "mydomain"
      logout_path: "/logout"
      redirect_uri: "https://mydomain.com/callback"
      scope: "openid"
      session:
        secret: <cannot-sub-this-one> 
    secretRef: apisix-plugin-openid
```

Would be great to hear anything about the status of this and whether any help is needed.

**Comment by tao12345666333** [comment]
It seems to be a fairly common requirement, and I have already made modifications to its status.
It will be implemented in v1.8.
